### PR TITLE
Pickling inferrers, changing directory structure, updating readme

### PR DESCRIPTION
**Pull request by akumgill**
* Adding more intermediate files (i.e. pickled inferrers)
* Updating directory structure so fewer nested directories
* Outputting files to output directory instead of to the config directory
* Updating script running flow to use config IDs instead of filenames
* Updated readme with new running instructions

### CODE REVIEW

**Comment by akumgill** [comment]
@cavestruz for review

**Review by cavestruz — CHANGES_REQUESTED** [review]
Ah, this one has similar merge conflicts, to one of the previous PRs I think.Can approve once those are resolved.

**Comment by akumgill** [comment]
resolved merge conflicts

File: README.rst [hunk 1]
```diff
@@ -3,46 +3,40 @@ weaklensingclustersbi
...
 ---------------
 
 Summary:
 
 We've split our workflow into four parts:
 
-1) Simulation (/configs/simulations) - sim_config tells us about how our simulations were generated/obtained (e.g. the mass-concentration relation used, the log10mass range, the number of simulations, the noise dex, etc).
-2) Observation(/configs/observations) - obs_config tells us about how our observations were generated/obtained (e.g. the richness-mass relation used, the richness range, the number of observations, the noise dex, etc). 
-3) Inference (/configs/inference) - infer_config tells us about how our inference was done (e.g. are we using SBI or MCMC, our priors, etc.)
+1) Simulation (/configs/simulations) - sim_config tells us about how our simulations were generated/obtained (e.g. the mass-concentration relation used, how much scatter the m-c relation has, the log10mass range, etc).
```
Comment (cavestruz) [inline README.rst:12]: These are great!  Can you also put together and put in the readme a visualization of the workflow in a pipeline diagram?

File: README.rst [hunk 2]
```diff
@@ -3,46 +3,40 @@ weaklensingclustersbi
...
+	This will output to the ``outputs/posteriors/{SIM_ID}.{INFER_ID}.{NUM_SIMS}`` directory
+
+3a) Create an observation config in the ``configs/observations`` directory
+3b) Generate observations using the following script: 
+	``python examples/gen_observations.py --obs_id {OBS_ID} --num_obs {NUM_OBS}``. 
+	This will output to the ``outputs/observations/{OBS_ID}.{NUM_OBS}`` directory
+
+4) Run inference using the following script:
+	``python examples/run_inference.py --sim_id {SIM_ID} --infer_id {INFER_ID} --obs_id {OBS_ID} --num_sims {NUM_SIMS} --num_obs {NUM_OBS}``.
+	This will output to the ``outputs/inference/{SIM_ID}.{INFER_ID}.{OBS_ID}.{NUM_SIMS}.{NUM_OBS}`` directory
```
Comment (cavestruz) [inline README.rst:39]: Nice job on the explicit run-time documentation :-)

**Review by cavestruz — CHANGES_REQUESTED** [review]
One request for a pipeline diagram to illustrate the code functionality for the README.

**Comment by akumgill** [comment]
added pipeline flowchart to readme

pipeline_flowchart

**Review by cavestruz — APPROVED** [review]
Flow chart looks good :-)

**Pull request merged**